mheimbinder/AirCastingiOS
# Congifure fixed session

**NOTE:** 
Apple doesn't offer an option to get names of nearby WiFi networks. Right now, the app gets the name of the network that the phone is connected with. We use this name on the popup and ask for a password. If the user wants to connect with a different WiFi, the text field for the network's name appears. 
It's not a perfect solution but I did my best to find a way to get the network's name. It's just not possible. We can try to figure out a different solution in the future.

_Also, somethings are temporary solutions and needs to be changed:_ 

- The progress line doesn't work yet in mobile session flow. 
- There's no error if the user won't add the session name.
- Even if the user selects an indoor session there'll be a location displayed on the "Are you ready" view. 
- Unfortunately, the new session won't be presented anywhere in the app. 

How to test:
1. Start creating session by choosing "+" tab bar button 
2. Choose session type: 
if FIXED
3. You should see "turn on Bluetooth"  and then "Power your AirBeam" views. 
if MOBILE
3. You should be able to select Bluetooth device or phone microphone (please, select the Bluetooth device). Then you should see "turn on Bluetooth"  and then "Power your AirBeam".

4. Now you enter the view where you can choose the device you want to connect with. Devices should be divided into two sections: AirBeams and others.  
You can refresh scanning for devices by clicking the "Refresh" button at the bottom of the view. When the phone is scanning for nearby devices, there should be two spinners displayed next to the section titles. 
Choose the device and click the "connect" button. 

5. You should be redirected to the "AirBeam Connecting" view and then, after few seconds "AirBeam Connected" view. 

6. Now you can add Session name and tags, choose AirBream placement and data transmission type. 

- If you choose indoor, the app will use send the AirBeam fake location. 
- If you choose the Outdoor app will get the current location and send it to AirBeam. 
- If you choose cellular - nothing changes (we didn't cover the cellular part of the fixed session). 
- If you choose wifi there's a popup where you can type the password for your network. 
- You can also choose to connect with a different WiFi network by clicking on the text below. If you do so, there's an additional text field for the network's name. 

7. In the last view you should see your session's name in the text and your location on the map below.
